src/main/kotlin/org/rust/ide/console/RsConsoleToolWindow.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.console
+
+import com.intellij.execution.ui.RunContentDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.SimpleToolWindowPanel
+import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.openapi.wm.ToolWindowManager
+import com.intellij.openapi.wm.ex.ToolWindowManagerListener
+import com.intellij.openapi.wm.impl.content.ToolWindowContentUi
+import com.intellij.ui.content.Content
+import com.intellij.ui.content.ContentFactory
+
+class RsConsoleToolWindow(private val project: Project) {
+
+    private var isInitialized: Boolean = false
+    private var isStateListenerInitialized: Boolean = false
+
+    fun initStateListener() {
+        if (isStateListenerInitialized) return
+        isStateListenerInitialized = true
+
+        project.messageBus.connect().subscribe(ToolWindowManagerListener.TOPIC, object : ToolWindowManagerListener {
+            override fun stateChanged() {
+                val toolWindow = toolWindow ?: return
+
+                val visible = toolWindow.isVisible
+                if (visible && toolWindow.contentManager.contentCount == 0) {
+                    val runner = RsConsoleRunnerFactory.getInstance().createConsoleRunner(project, null)
+                    runner.run(true)
+                }
+            }
+        })
+    }
+
+    fun setContent(contentDescriptor: RunContentDescriptor) {
+        val toolWindow = toolWindow!!
```

Comment:
Please try to avoid `!!`
```suggestion
        val toolWindow = toolWindow ?: return
```

Reply:
Another option: `toolWindow ?: error("explain why it can't be null")`